graylog2-web-interface/src/components/graylog/OverlayTrigger.tsx

```diff
@@ -55,15 +71,40 @@ class OverlayTrigger extends React.Component<Props, State> {
   }
 
   render() {
-    const { children, container, placement, overlay, rootClose, ...overlayProps } = this.props;
+    const { children, container, placement, overlay, rootClose, trigger, ...overlayProps } = this.props;
     const { show } = this.state;
 
-    const toggleShow = () => this.setState({ show: !show });
+    const setShow = (isShown: boolean) => this.setState({ show: isShown });
+    const toggleShow = () => setShow(!show);
+
+    const _onTrigger = {
+      click: {
+        onClick: toggleShow,
+      },
+      hover: {
+        onMouseEnter: () => setShow(true),
+        onMouseLeave: () => setShow(false),
+      },
+      focus: {
+        onFocus: () => setShow(true),
+        onBlur: () => setShow(false),
+      },
+    };
+
+    let triggerables = _onTrigger[trigger];
+
+    if (Array.isArray(trigger)) {
+      const multipleTriggers = {};
+
+      trigger.forEach((triggerType) => {
+        triggerables = assign(multipleTriggers, _onTrigger[triggerType]);
```

Comment:
Is there a reason that you are not using `Object.assign` here?
```suggestion
        triggerables = Object.assign(multipleTriggers, _onTrigger[triggerType]);
```